subgraph-js/src/printSubgraphSchema.ts

```diff
@@ -104,6 +104,10 @@ function printSchemaDefinition(schema: GraphQLSchema): string | undefined {
     operationTypes.push(`  subscription: ${subscriptionType.name}`);
   }
 
+  if (operationTypes.length === 0) {
```

Comment:
Is this change necessary or related? I don't see a new or affected test to understand why it's been added. Not every subgraph must provide an entry point via a root operation (i.e. subgraphs can just extend types and nothing more) so I don't think this change is correct.

Reply:
This is only needed in the case where your schema-definition has no directives (e.g. `@contact`), and where your schema does not define types for `mutation` or `subscription`, or a type for `query` other than `Query`. Removing this line will cause the test I added to break for example.

Reply:
I would say that the test case is invalid since you're referencing a type `Query` which you haven't defined. I think this approach falls short in the case that a subgraph _shouldn't_ have a schema definition at all.

Reply:
Good shout, and I actually managed to get rid of one of the two pieces of code that you didn't think were necessary by adding a `Query` type 👍 .